Saved feature file (.fea) disagrees with .sfd

When saving out an Adobe feature file, FontForge is too aggressive in grouping together "anchor-equal" glyphs, i.e. glyphs with the same base-to-mark anchors. Or, another way of putting it: FontForge's criteria for the "anchor-equality" of two glyphs is too weak/lenient. A single equal anchor (as long as it is found before any unequal anchor) is taken to constitute "anchor equality" of the two glyphs.

- [x] The FontForge version and the operating system you're using
Version: 20200314
Based on sources from 2020-03-14 03:45 UTC-ML-TtfDb-D-GDK3.
Based on source from git with hash: c94f19e71cdbea2272ed6508f2d38a5b060dca9e
- [x] The behavior you expect to see, and the actual behavior
Excerpt of expected behavior (note `pedagesh` has value 640 and is not lumped in with `samekhdagesh` & `kofdagesh`, which share value 570):
```
lookup markUpperMarksPositioning {
    markClass [\pashta ] <anchor 309 0> @A29;
    pos base \pedagesh <anchor 640 0> mark @A29
    pos base [\samekhdagesh \kofdagesh ] <anchor 570 0> mark @A29
} markUpperMarksPositioning;
```
I derive this expectation from the following excerpt from the `.sfd`:
```
StartChar: pashta
AnchorPoint: "A29" 309 0 mark 0
EndChar

StartChar: pedagesh
AnchorPoint: "A29" 640 0 basechar 0
EndChar
```
And my expectation is further strengthened by the following excerpt from a `.fea` decompilation produced by running  `otf2fea` on the `.ttf` generated by FontForge (`otf2fea` written by @simoncozens):
```
markClass pashta <anchor 309 0> @Anchor48;
lookup MarkToBase41 {
    pos base pedagesh <anchor 640 0> mark @Anchor48;
} MarkToBase41;
```
Excerpt of actual behavior (note `pedagesh` is lumped in with `samekhdagesh` & `kofdagesh`, with all 3 sharing value 570):
```
lookup markUpperMarksPositioning {
    markClass [\pashta ] <anchor 309 0> @A29;
    pos base [\samekhdagesh \pedagesh \kofdagesh ] <anchor 570 0> mark @A29
} markUpperMarksPositioning;
```
- [x] Steps to reproduce the behavior
Choose "Save feature file" for the attached `.sfd`: [TaameyFrankCLM-Medium-0d114.zip](https://github.com/fontforge/fontforge/files/6446590/TaameyFrankCLM-Medium-0d114.zip)

Thanks @skef I will try to put together a pull request.

First hurdle is I think I followed the build instructions in INSTALL.md but on the compile of every file I'm getting:

> fatal error: fontforge-config.h: No such file or directory.

BTW I also checked the instructions in "[CMake guide for FontForge](https://github.com/fontforge/fontforge/wiki/CMake-guide-for-FontForge)" and they seem to agree with the instructions in INSTALL.md.

On running the following:

```
cd fontforge
mkdir build
cd build
cmake -GNinja ..
ninja
ninja install
```

I get the following output (excerpted)

> [... Lots of CMake stuff ... ]
> -- Generating done
> -- Build files have been written to: /home/bdenckla/fontforge/fontforge/build
> [1/95] Building C object CMakeFiles/fontforge.dir/autohint.o
> FAILED: CMakeFiles/fontforge.dir/autohint.o
> /usr/bin/cc  -I../  -MD -MT CMakeFiles/fontforge.dir/autohint.o -MF CMakeFiles/fontforge.dir/autohint.o.d -o CMakeFiles/fontforge.dir/autohint.o   -c ../autohint.c
> ../autohint.c:28:10: fatal error: fontforge-config.h: No such file or directory
>    28 | #include <fontforge-config.h>
>       |          ^~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> [2/95] Building C object CMakeFiles/fontforge.dir/activeinui.o
> FAILED: CMakeFiles/fontforge.dir/activeinui.o
> /usr/bin/cc  -I../  -MD -MT CMakeFiles/fontforge.dir/activeinui.o -MF CMakeFiles/fontforge.dir/activeinui.o.d -o CMakeFiles/fontforge.dir/activeinui.o   -c ../activeinui.c
> ../activeinui.c:28:10: fatal error: fontforge-config.h: No such file or directory
>    28 | #include <fontforge-config.h>
>       |          ^~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> [Many more /usr/bin/cc failures like those above]
> compilation terminated.
> ninja: build stopped: subcommand failed.
> ninja: error: unknown target 'install'

Is `fontforge-config.h` a header file generated by CMake, the way good old `configure` does in the GNU autotools build system?

--- a/fontforge/featurefile.c
+++ b/fontforge/featurefile.c
@@ -1017,7 +1017,7 @@ return( true );
 return( false );
 }
 
-static int SameBaseAP(SplineChar *sc1,SplineChar *sc2,struct lookup_subtable *sub) {
+static int AsymmetricSameBaseAP(SplineChar *sc1,SplineChar *sc2,struct lookup_subtable *sub) {
     AnchorPoint *ap1, *ap2;
 
     for ( ap1=sc1->anchor; ap1!=NULL; ap1=ap1->next ) {
@@ -1034,10 +1034,15 @@ return( false );
 	    break;
 		}
 	    }
-return( ap2!=NULL );
+	    if ( ap2==NULL )
+return( false );
 	}
     }
-return( false );
+return( true );
+}
+
+static int SameBaseAP(SplineChar *sc1,SplineChar *sc2,struct lookup_subtable *sub) {
+    return AsymmetricSameBaseAP(sc1,sc2,sub) && AsymmetricSameBaseAP(sc2,sc1,sub);
 }
 
 static void dump_anchors(FILE *out,SplineFont *sf,struct lookup_subtable *sub) {

